x/common/testutil/action/testcase.go

```diff
@@ -48,67 +48,73 @@ func TC(name string) TestCase {
 	return TestCase{Name: name}
 }
 
-func (t TestCase) Given(action ...Action) TestCase {
-	t.given = append(t.given, action...)
-	return t
+func (tc TestCase) Given(action ...Action) TestCase {
+	tc.given = append(tc.given, action...)
+	return tc
 }
 
-func (t TestCase) When(action ...Action) TestCase {
-	t.when = append(t.when, action...)
-	return t
+func (tc TestCase) When(action ...Action) TestCase {
+	tc.when = append(tc.when, action...)
+	return tc
 }
 
-func (t TestCase) Then(action ...Action) TestCase {
-	t.then = append(t.then, action...)
-	return t
+func (tc TestCase) Then(action ...Action) TestCase {
+	tc.then = append(tc.then, action...)
+	return tc
 }
 
-type TestSuite struct {
-	t *testing.T
-
-	testCases []TestCase
-}
-
-func NewTestSuite(t *testing.T) *TestSuite {
-	return &TestSuite{t: t}
-}
-
-func (t *TestSuite) WithTestCases(testCase ...TestCase) *TestSuite {
-	t.testCases = append(t.testCases, testCase...)
-	return t
-}
-
-func (t *TestSuite) Run() {
-	for _, testCase := range t.testCases {
+func (tc TestCase) Run(t *testing.T) {
+	t.Run(tc.Name, func(t *testing.T) {
```

Comment:
All that's happening here is that the execution block is wrapped inside of a `t.Run`. This makes the test logs less opaque because it clarifies further which specific test case is failing. 

One issue with out stack trace right now is that it always shows a failing test as coming from `testcase.go`.